src/libs/actions/Policy.js

```diff
@@ -88,6 +90,15 @@ function getPolicyList() {
                     },
                 }), {});
                 Onyx.mergeCollection(ONYXKEYS.COLLECTION.POLICY, policyDataToStore);
```

Comment:
I'm actually thinking that maybe this whole PR can be simplified by just changing this line to:

```javascript
Onyx.mergeCollection(ONYXKEYS.COLLECTION.POLICY, {
    // Erase all policies in Onyx
    ...(_.map(_.keys(allPolicies), key => ({[key]: null}))),

    // And overwrite them with only the ones returned by the API call
    ...policyDataToStore,
});
```

That has a few advantages:

1. Only one Onyx write instead of two
1. Less concerned with the inner workings of Onyx
1. In my opinion a bit simpler to understand

If this is a pattern we keep seeing (i.e: replace a full collection instead of merging a new collection into an existing one), we could abstract it into a utility function within Onyx called `setCollection` or `overwriteCollection`.

Reply:
Sorry, made a couple edits to my code snippet there but the above is working ^

Reply:
> If this is a pattern we keep seeing (i.e: replace a full collection instead of merging a new collection into an existing one), we could abstract it into a utility function within Onyx called setCollection or overwriteCollection.

That was my initial approach, but, after discussing it with Marc, we are not creating the function for now: https://expensify.slack.com/archives/C03TQ48KC/p1628248835259300

I like your single Onyx call idea, and it tests well. Updated!